packages/react-components/src/icons/src/Icon.jsx

```diff
@@ -15,18 +15,28 @@ const propTypes = {
      * An icon can vary in size.
      */
     size: oneOf(["2xs", "xs", "sm", "md", "lg", "xl", "inherit"]),
+    /**
+     * An icon can change inherit it's parent color.
+     */
+    color: oneOf(["inherit"]),
     /**
      * Default slot override.
      */
     slot: string
 };
 
+export function getIconClass(color) {
+    const colorClass = color ? `color-${color}`: "";
+
+    return colorClass;
```

Comment:
That type of code should be directly in the `cssModule`.  I understand you duplicate it twice but it is super simple code.